src/core/server/elasticsearch/client/retry_call_cluster.ts

```diff
@@ -0,0 +1,103 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { defer, throwError, iif, timer } from 'rxjs';
+import { concatMap, retryWhen } from 'rxjs/operators';
+import { Logger } from '../../logging';
+
+const retryResponseStatuses = [
+  503, // ServiceUnavailable
```

Comment:
Note, if the status code of a response is `502`, `503` or `504`, the client performs an automatic failover [internally](https://github.com/elastic/elasticsearch-js/blob/1592c4d5756d21f7d1463795b39cab3af1049c8a/lib/Transport.js#L222), so have it here could be a duplication.
Regarding `4xx` status codes, the client does not perform any automatic failover/retry.

Reply:
Good to know. Having this feature for our 'end' consumers is great. For the migration needs, we need to perform an infinity of retries, so keeping the logic in a single place is probably better. But we don't need to disable the internal failover here though, the duplication is probably fine.